### PR TITLE
Stelau/jest

### PR DESCRIPTION
**Pull request by toshi38**
Switch to jest removing a selection of inferior test frameworks and also sinon since we don't need it anymore.

### CODE REVIEW

File: src/specs/component.app.spec.ts [hunk 1]
```diff
@@ -42,8 +40,11 @@ let debugElement: DebugElement
 let nativeElement: HTMLElement;
 
 describe('AppComponent', () => {
-  before(() => TestBed.configureTestingModule(testModuleConfig));
-  after(() => getTestBed().resetTestingModule());
+  beforeEach(() => TestBed.configureTestingModule(testModuleConfig));
```
Comment (krawaller) [inline src/specs/component.app.spec.ts:43]: Do we really need to do this before each test? The config doesn't change, does it?

File: src/specs/component.combatant.spec.ts [hunk 2]
```diff
@@ -10,14 +10,13 @@ Unit tests for the Combatant component. We need to test...
...
-import * as sinon from 'sinon';
 
 const fakeBattleServiceObservable = {
-  subscribe: sinon.stub()
+  subscribe: jest.fn()
 };
 
 const fakeBattleService = {
-  battleInfoForUser: sinon.stub().returns(fakeBattleServiceObservable)
+  battleInfoForUser: jest.fn().mockReturnValue(fakeBattleServiceObservable)
```
Comment (krawaller) [inline src/specs/component.combatant.spec.ts:19]: Can also just do `jest.fn(() => fakeBattleServiceObservable)` (i think)

File: src/specs/component.app.spec.ts (hunk 1 above)
Comment (toshi38) [reply src/specs/component.app.spec.ts:43]: not sure off the top of my head...

**Review by krawaller — APPROVED** [review]
Less setup, less hassle! Nice!

File: src/specs/component.combatant.spec.ts (hunk 2 above)
Comment (toshi38) [reply src/specs/component.combatant.spec.ts:19]: maybe :)

File: src/specs/component.app.spec.ts (hunk 1 above)
Comment (toshi38) [reply src/specs/component.app.spec.ts:43]: seems to be needed in a before each.

**Pull request merged**